addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/protocol/RemoteControllerWebSocket.java

```diff
@@ -0,0 +1,678 @@
+ *
+ * @author Arjan Mels - Initial contribution
+ */
+public class RemoteControllerWebSocket extends RemoteController implements Listener {
+
+    private final Logger logger = LoggerFactory.getLogger(RemoteControllerWebSocket.class);
+
+    private final static String WS_ENDPOINT_REMOTE_CONTROL = "/api/v2/channels/samsung.remote.control";
+    private final static String WS_ENDPOINT_ART = "/api/v2/channels/com.samsung.art-app";
+    private final static String WS_ENDPOINT_V2 = "/api/v2";
+    private WebSocketClient client;
+
+    UUID uuid = UUID.randomUUID();
+
+    private RemoteControllerWebsocketCallback callback;
+
+    WebSocketRemote webSocketRemote = new WebSocketRemote();
+    WebSocketArt webSocketArt = new WebSocketArt();
+    WebSocketV2 webSocketV2 = new WebSocketV2();
+
+    /**
+     * Create and initialize remote controller instance.
+     *
+     * @param host                    Host name of the Samsung TV.
+     * @param port                    TCP port of the remote controller protocol.
+     * @param appName                 Application name used to send key codes.
+     * @param uniqueId                Unique Id used to send key codes.
+     * @param remoteControllerService
+     */
+
+    public RemoteControllerWebSocket(String host, int port, String appName, String uniqueId,
+            RemoteControllerWebsocketCallback remoteControllerWebsocketCallback) {
+        super(host, port, appName, uniqueId);
+        this.client = new WebSocketClient();
+        this.client.addLifeCycleListener(this);
+
+        this.callback = remoteControllerWebsocketCallback;
+    }
+
+    private boolean isConnected() {
+        return webSocketRemote.isConnected();
+    }
+
+    @Override
+    public void openConnection() throws RemoteControllerException {
+        if (!(client.isStarted() || client.isStarting())) {
+            logger.debug("RemoteControllerWebSocket start Client");
+            try {
+                client.start();
+                client.setMaxBinaryMessageBufferSize(1000000);
+                // websocket connect will be done in lifetime handler
+                return;
+
+            } catch (Exception e) {
+                logger.warn("Cannot connect to websocket remote control interface: " + e.getMessage());
+                throw new RemoteControllerException(e);
+            }
+        }
+        connectWebSockets();
+    }
+
+    private void connectWebSockets() {
+        try {
+            webSocketRemote.connect(new URI("ws", null, host, port, WS_ENDPOINT_REMOTE_CONTROL, "name=openhab", null));
+        } catch (RemoteControllerException | URISyntaxException e) {
+            logger.warn("Problem connecting to remote websocket", e);
+        }
+        try {
+            webSocketArt.connect(new URI("ws", null, host, port, WS_ENDPOINT_ART, "name=openhab", null));
+        } catch (RemoteControllerException | URISyntaxException e) {
+            logger.warn("Problem connecting to artmode websocket", e);

```

Comment:
Should we report this through the status of the Thing, if all of them fail I would expect something to be OFFLINE, if just one or two fail you could maybe set at least a message in the status of the Thing.

Reply:
can you address this comment?